### PR TITLE
CTAP2.1 improvements

### PR DESCRIPTION
**Pull request by AdamVe**
Each commit fixes a specific issue with my CTAP2.1 implementation.

### CODE REVIEW

File: fido/src/main/java/com/yubico/yubikit/fido/client/BasicWebAuthnClient.java [hunk 1]
```diff
@@ -601,6 +592,27 @@ private boolean getCtapUv(String userVerification, boolean pinProvided) throws C
         }
     }
 
+    /**
+     * Calculates the preferred pinUvAuth protocol version for authenticator provided list.
+     * Returns PinUvAuthDummyProtocol.VERSION if the authenticator does not support any of the SDK
+     * supported protocols.
+     */
+    private int getPreferredPinUvAuthProtocol(List<Integer> pinUvAuthProtocols) {
```
Comment (dainnilsson) [inline fido/src/main/java/com/yubico/yubikit/fido/client/BasicWebAuthnClient.java:614]: Would it make more sense to return an instance of PinUvAuthProtocol?

Comment (AdamVe) [reply fido/src/main/java/com/yubico/yubikit/fido/client/BasicWebAuthnClient.java:614]: Good suggestion, I will update.

Comment (AdamVe) [reply fido/src/main/java/com/yubico/yubikit/fido/client/BasicWebAuthnClient.java]: fixed

**Review by dainnilsson — APPROVED** [review]

File: fido/src/main/java/com/yubico/yubikit/fido/client/BasicWebAuthnClient.java [hunk 2]
```diff
@@ -607,24 +604,22 @@ private boolean getCtapUv(String userVerification, boolean pinProvided) throws C
...
-                    return supportedProtocol;
-                }
+            if (pinSupported && protocol == PinUvAuthProtocolV1.VERSION) {
+                return new PinUvAuthProtocolV1();
+            }
+
+            if (pinSupported && protocol == PinUvAuthProtocolV2.VERSION) {
+                return new PinUvAuthProtocolV2();
             }
         }
```
[the remark this answers is not in the document]
Comment (AdamVe) [reply fido/src/main/java/com/yubico/yubikit/fido/client/BasicWebAuthnClient.java:620]: will fix & merge

**Pull request merged**